app/src/main/java/com/example/fitjournal/core/presentation/commoncomponents/dialogs/components/editWorkoutSet/EditWorkoutSetCalisthenics.kt

```diff
@@ -0,0 +1,256 @@
+            isWorkoutValid(false, null)
+        } else {
+            isWorkoutValid(
+                true,
+                WorkoutTypeDialog.Calisthenics(
+                    editWorkoutSetCalisthenicsModel = EditWorkoutSetCalisthenicsModel(
+                        reps = repValue,
+                        sets = setValue,
+                        weight = weightValue,
+                        weightType = weightType,
+                        hr = hourValue,
+                        min = minuteValue,
+                        sec = secondValue,
+                        index = editWorkoutSetCalisthenicsModel.index
+                    )
+                )
+            )
+        }
+    }
+    LaunchedEffect(
+        key1 = hourValue,
+        key2 = minuteValue,
+        key3 = secondValue
+    ) {
+        if (isRepErrorVisible || isSetErrorVisible || isTimeErrorVisible || isWeightErrorVisible) {
+            isWorkoutValid(false, null)
+        } else {
+            isWorkoutValid(
+                true,
+                WorkoutTypeDialog.Calisthenics(
+                    editWorkoutSetCalisthenicsModel = EditWorkoutSetCalisthenicsModel(
+                        reps = repValue,
+                        sets = setValue,
+                        weight = weightValue,
+                        weightType = weightType,
+                        hr = hourValue,
+                        min = minuteValue,
+                        sec = secondValue,
+                        index = editWorkoutSetCalisthenicsModel.index
+                    )
+                )
+            )
+        }
+    }
+
+    EditWorkoutSetsSection(
+        isErrorVisible = isSetErrorVisible,
+        editSetsValue = { editWorkoutFunction: EditWorkoutFunction, sets: String ->
+            isSetErrorVisible = false
+            try {
+                if (sets.isDigitsOnly() && sets != "0" && sets.isNotEmpty()) {
+                    var setsInt = sets.toIntOrNull()
+                    if (setsInt == null) {
+                        isSetErrorVisible = true
+                        return@EditWorkoutSetsSection
+                    }
+                    when (editWorkoutFunction) {
+                        EditWorkoutFunction.ADD_VALUE -> setsInt += 1
+                        EditWorkoutFunction.SUBTRACT_VALUE -> setsInt -= 1
+                    }
+                    setValue = setsInt.toString()
+                } else if (sets == "0" || sets.isEmpty()) {
+                    setValue = when (editWorkoutFunction) {
+                        EditWorkoutFunction.ADD_VALUE -> "1"
+                        EditWorkoutFunction.SUBTRACT_VALUE -> "0"
+                    }
+                }
+                if (setValue == "0") {
+                    isSetErrorVisible = true
+                }
+            } catch (e: Exception) {
+                isSetErrorVisible = true
+            }
+        },
```

Comment:
This is duplicate code. Can we extract this out into a helper function and reuse it?